math/mathcore/inc/Math/FitMethodFunction.h

```diff
@@ -98,9 +98,24 @@ class BasicFitMethodFunction : public FunctionType {
 
 public:
 
+  /* traits for knowing if is a function supporting gradient*/
+  static bool IsAGradFCN() {
+     return ObjFCNTrait<FunctionType>::IsGrad();
+  }
```

Comment:
What's the advantage of using this `ObjFCNTrait` struct over Implementing `IsAGradFCN` directly? E.g. having this in the class definition:

```C++
   static bool IsAGradFCN() {
      return false;
   }
```
And then having this after:
```C++
template<>
inline bool BasicFitMethodFunction<ROOT::Math::IMultiGradFunction>::IsAGradFCN() {
   return true;
}
```

Reply:
Thanks, I did not think this could work. I will try...